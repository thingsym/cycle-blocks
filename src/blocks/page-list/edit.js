'use strict';

/**
 * WordPress dependencies
 */
import {
	PanelBody,
	TextControl,
	SelectControl,
	RangeControl,
	ToggleControl,
	Disabled,
	Placeholder,
} from '@wordpress/components';
import { useMemo } from '@wordpress/element';
import {
	InspectorControls,
	useBlockProps,
} from '@wordpress/block-editor';
import { __ } from '@wordpress/i18n';
import ServerSideRender from '@wordpress/server-side-render';

/**
 * Internal dependencies
 */
import FeaturedImageUploadControl from '../../helper/featured-image-upload-control/index.js';

export default function pageListEdit( { attributes, setAttributes } ) {
	const {
		postId,
		layout,
		sortColumn,
		sortOrder,
		displayAncestor,
		columns,
		depth,
		displayFeaturedImage,
		featuredImageId,
	} = attributes;

	const getLayoutOptions = useMemo( () => {
		const selectOption = [
			{
				label: __( 'List', 'cycle-blocks' ),
				value: 'list',
			},
			{
				label: __( 'Card', 'cycle-blocks' ),
				value: 'card',
			},
			{
				label: __( 'Topics', 'cycle-blocks' ),
				value: 'topics',
			},
			{
				label: __( 'Article', 'cycle-blocks' ),
				value: 'article',
			}
		];

		return selectOption;
	}, [] );

	const getSortColumnOptions = useMemo( () => {
		const selectOption = [
			{
				label: __( 'menu_order', 'cycle-blocks' ),
				value: 'menu_order'
			},
			{
				label: __( 'post_title', 'cycle-blocks' ),
				value: 'post_title'
			},
			{
				label: __( 'post_date', 'cycle-blocks' ),
				value: 'post_date'
			},
			{
				label: __( 'post_modified', 'cycle-blocks' ),
				value: 'post_modified'
			},
			{
				label: __( 'ID', 'cycle-blocks' ),
				value: 'ID'
			}
		];

		return selectOption;
	}, [] );

	const getSortOrderOptions = useMemo( () => {
		const selectOption = [
			{
				label: __( 'ASC', 'cycle-blocks' ),
				value: 'ASC'
			},
			{
				label: __( 'DESC', 'cycle-blocks' ),
				value: 'DESC'
			}
		];

		return selectOption;
	}, [] );

	const inspectorControls = (
		<InspectorControls>
			<PanelBody title={ __( 'Settings', 'cycle-blocks' ) } >
				<TextControl
					label={ __( 'Post ID', 'cycle-blocks' ) }
					value={ postId }
					onChange={ ( value ) =>
						setAttributes( {
							postId: value ? value : '',
						} )
					}
				/>
				<SelectControl
					label={ __( 'Layout', 'cycle-blocks' ) }
					options={ getLayoutOptions }
					value={ layout }
					onChange={ ( value ) =>
						setAttributes( {
							layout: value ? value : '',
						} )
					}
				/>
				<SelectControl
					label={ __( 'Sort Column', 'cycle-blocks' ) }
					options={ getSortColumnOptions }
					value={ sortColumn }
					onChange={ ( value ) =>
						setAttributes( {
							sortColumn: value ? value : '',
						} )
					}
				/>
				<SelectControl
					label={ __( 'Sort Order', 'cycle-blocks' ) }
					options={ getSortOrderOptions }
					value={ sortOrder }
					onChange={ ( value ) =>
						setAttributes( {
							sortOrder: value ? value : '',
						} )
					}
				/>
				{ layout === 'list' && (
					<ToggleControl
						label={ __( 'Display Ancestor', 'cycle-blocks' ) }
						checked={ displayAncestor }
						onChange={ () =>
							setAttributes( {
								displayAncestor: ! displayAncestor,
							} )
						}
					/>
				) }
				{ layout === 'list' && (
					<RangeControl
						label={ __( 'Depth', 'cycle-blocks' ) }
						value={ depth }
						onChange={ ( value ) =>
							setAttributes( {
								depth: value
							} )
						}
						min={ 1 }
						max={ 4 }
					/>
				) }
				{ layout === 'card' && (
					<RangeControl
						label={ __( 'Columns', 'cycle-blocks' ) }
						value={ columns }
						onChange={ ( value ) =>
							setAttributes( { columns: value } )
						}
						min={ 2 }
						max={ 4 }
						required
					/>
				) }
				{ layout !== 'list' && (
					<ToggleControl
						label={ __( 'Display featured image', 'cycle-blocks' ) }
						checked={ displayFeaturedImage }
						onChange={ () =>
							setAttributes( {
								displayFeaturedImage: ! displayFeaturedImage
							} )
						}
					/>
				) }
				{ layout !== 'list' && displayFeaturedImage && (
					<FeaturedImageUploadControl
						setAttributes={ setAttributes }
						value={ featuredImageId }
					/>
				) }
			</PanelBody>
		</InspectorControls>
	);

	return (
		<>
			{ inspectorControls }
			<div { ...useBlockProps() }>
				<Disabled>
					<ServerSideRender
						block="cycle-blocks/page-list"
						attributes={ attributes }
						EmptyResponsePlaceholder={ () => (
							<Placeholder>
							{ __( 'Not Found pages. If the Post ID is blank or not a post/page, it cannot be viewed in the editor.', 'cycle-blocks' ) }
							</Placeholder>
						) }
					/>
				</Disabled>
			</div>
		</>
	);
}
