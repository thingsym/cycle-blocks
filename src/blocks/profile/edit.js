'use strict';

/**
 * External dependencies
 */
import { get, forEach, invoke } from 'lodash';
import classnames from 'classnames';

/**
 * WordPress dependencies
 */
import { RawHTML } from '@wordpress/element';
 import {
	BlockControls,
	InspectorControls,
	RichText,
	useBlockProps,
} from '@wordpress/block-editor';
import {
	PanelBody,
	SelectControl,
	ToggleControl,
	RangeControl,
	ToolbarGroup,
	Placeholder,
} from '@wordpress/components';
import { dateI18n, format, getSettings } from '@wordpress/date';
import { useSelect } from '@wordpress/data';
import { __ } from '@wordpress/i18n';
import { store as coreStore } from '@wordpress/core-data';
import { pin, list, grid } from '@wordpress/icons';

/**
 * Internal dependencies
 */
import metadata from './block.json';
import UserControl from '../../helper/user-control/index.js';
import FeaturedImageUploadControl from '../../helper/featured-image-upload-control/index.js';

const { title: metadataTitle } = metadata;

const MAX_POSTS_COLUMNS = 6;

export default function profileEdit( {
	isSelected,
	context: { postType, postId, queryId },
	attributes,
	setAttributes,
} ) {
	const {
		userId,
		showTitle,
		title,
		showAvatar,
		showBio,
		showRecentPosts,
		authorTitle,
		postsToShow,
		displayFeaturedImage,
		featuredImageId,
		postLayout,
		postColumns,
		byline
	} = attributes;

	const { authorId, authorDetails } = useSelect(
		( select ) => {
			const { getEditedEntityRecord, getUser } = select( coreStore );

			const _authorId = ! userId ? getEditedEntityRecord(
				'postType',
				postType,
				postId
			)?.author : null;

			return {
				authorId: userId ? userId : _authorId,
				authorDetails: userId ? getUser( userId ) : getUser( _authorId ),
			};
		},
		[
			userId,
		]
	);

	const latestPosts = useSelect(
		( select ) => {
			if ( ! showRecentPosts ) {
				return {};
			}

			const { getEntityRecords, getMedia } = select( coreStore );

			const posts = getEntityRecords(
				'postType',
				'post',
				{
					author: authorId,
					order: 'desc',
					orderby: 'date',
					per_page: postsToShow,
				}
			);

			const latestPosts = ! Array.isArray( posts )
				? posts
				: posts.map( ( post ) => {
					if ( ! post.featured_media ) return post;

					const image = getMedia( post.featured_media );
					let url = get(
						image,
						[
							'media_details',
							'sizes',
							'thumbnail',
							'source_url',
						],
						null
					);
					if ( ! url ) {
						url = get( image, 'source_url', null );
					}
					const featuredImageInfo = {
						url,
						// eslint-disable-next-line camelcase
						alt: image?.alt_text,
					};
					return { ...post, featuredImageInfo };
				} );

			return latestPosts;
		},
		[
			userId,
			authorId,
			showRecentPosts,
			postsToShow,
			displayFeaturedImage
		]
	);

	const avatarSizes = [];
	if ( authorDetails ) {
		forEach( authorDetails.avatar_urls, ( url, size ) => {
			avatarSizes.push( {
				value: size,
				label: `${ size } x ${ size }`,
			} );
		} );
	}

	const layoutControls = [
		{
			icon: list,
			title: __( 'List view', 'cycle-blocks' ),
			onClick: () => setAttributes( { postLayout: 'list' } ),
			isActive: postLayout === 'list',
		},
		{
			icon: grid,
			title: __( 'Grid view', 'cycle-blocks' ),
			onClick: () => setAttributes( { postLayout: 'grid' } ),
			isActive: postLayout === 'grid',
		},
	];

	const dateFormat = getSettings().formats.date;

	const blockProps = useBlockProps( {
		className: 'cycle-blocks-profile'
	} );

	const post_class = classnames( {
		'cycle-blocks-profile__recent-post--layout-grid': postLayout === 'grid',
		[ `cycle-blocks-profile__recent-post--columns-${ postColumns }` ]: postLayout === 'grid',
	} );

	const { mediaSizes } = useSelect(
		( select ) => {
			if ( ! featuredImageId ) {
				return {};
			}

			const { getMedia } = select( coreStore );
			const mediaObj = getMedia( featuredImageId, { context: 'view' } );

			return {
				media: mediaObj,
				mediaSizes: mediaObj ? mediaObj.media_details.sizes : undefined,
			};
		},
		[ featuredImageId ]
	);

	const hasPosts = !! latestPosts?.length;

	const inspectorControls = (
		<InspectorControls>
			<PanelBody title={ __( 'Profile Settings', 'cycle-blocks' ) }>
				<UserControl
					value={ attributes?.userId }
					onChange={ ( value ) => {
						setAttributes( {
							userId: value,
						} );
					} }
				/>

				<ToggleControl
					label={ __( 'Show title', 'cycle-blocks' ) }
					checked={ showTitle }
					onChange={ () =>
						setAttributes( { showTitle: ! showTitle } )
					}
				/>
				<ToggleControl
					label={ __( 'Show avatar', 'cycle-blocks' ) }
					checked={ showAvatar }
					onChange={ () =>
						setAttributes( { showAvatar: ! showAvatar } )
					}
				/>
				{ showAvatar && (
					<SelectControl
						label={ __( 'Avatar size', 'cycle-blocks' ) }
						value={ attributes.avatarSize }
						options={ avatarSizes }
						onChange={ ( size ) => {
							setAttributes( {
								avatarSize: Number( size ),
							} );
						} }
					/>
				) }
				<ToggleControl
					label={ __( 'Show bio', 'cycle-blocks' ) }
					checked={ showBio }
					onChange={ () =>
						setAttributes( { showBio: ! showBio } )
					}
				/>
				<ToggleControl
					label={ __( 'Show recent posts', 'cycle-blocks' ) }
					checked={ showRecentPosts }
					onChange={ () =>
						setAttributes( { showRecentPosts: ! showRecentPosts } )
					}
				/>

				{ showRecentPosts && (
					<RangeControl
						label={ __( 'Number of Posts', 'cycle-blocks' ) }
						value={ postsToShow }
						onChange={ ( value ) =>
							setAttributes( { postsToShow: value } )
						}
						min={ 1 }
						max={ 24 }
						required
					/>
				) }
				{ showRecentPosts && postLayout === 'grid' && (
					<RangeControl
						label={ __( 'Columns', 'cycle-blocks' ) }
						value={ postColumns }
						onChange={ ( value ) =>
							setAttributes( { postColumns: value } )
						}
						min={ 3 }
						max={
							! hasPosts
								? MAX_POSTS_COLUMNS
								: Math.min(
										MAX_POSTS_COLUMNS,
										latestPosts.length
									)
						}
						required
					/>
				) }
				{ showRecentPosts && (
					<ToggleControl
						label={ __( 'Display featured image', 'cycle-blocks' ) }
						checked={ displayFeaturedImage }
						onChange={ () =>
							setAttributes( { displayFeaturedImage: ! displayFeaturedImage } )
						}
					/>
				) }
				{ showRecentPosts && displayFeaturedImage && (
					<FeaturedImageUploadControl
						value={ featuredImageId }
						setAttributes={ setAttributes }
					/>
				) }
			</PanelBody>
		</InspectorControls>
	);

	if ( ! authorId ) {
		return (
			<div { ...blockProps }>
				{ inspectorControls }
				{ showRecentPosts && (
					<BlockControls>
						<ToolbarGroup controls={ layoutControls } />
					</BlockControls>
				) }
				<Placeholder
					icon={ pin }
					label={ metadataTitle }
				>
					{ __( 'Not Found authorId. If the User is blank or not a post/page, it cannot be viewed in the editor.', 'cycle-blocks' ) }
				</Placeholder>
			</div>
		);
	}

	return (
		<>
			{ inspectorControls }
			{ showRecentPosts && (
				<BlockControls>
					<ToolbarGroup controls={ layoutControls } />
				</BlockControls>
			) }
			<div { ...blockProps }>
				{ showTitle && (
					<div className="cycle-blocks-profile__header">
						{ ( ! RichText.isEmpty( title ) || isSelected ) && (
							<RichText
								className="cycle-blocks-profile__title"
								tagName={ "h3" }
								multiline={ false }
								aria-label={ __( 'Title', 'cycle-blocks' ) }
								placeholder={ __( 'Write title…', 'cycle-blocks' ) }
								value={ title }
								onChange={ ( value ) =>
									setAttributes( { title: value } )
								}
							/>
						) }
					</div>
				) }
				{ showAvatar && authorDetails && (
					<div className="cycle-blocks-profile__avatar">
						<img
							width={ attributes.avatarSize }
							src={
								authorDetails.avatar_urls[
									attributes.avatarSize
								]
							}
							alt={ authorDetails.name }
						/>
					</div>
				) }
				<div className="cycle-blocks-profile__content">
					{ ( ! RichText.isEmpty( byline ) || isSelected ) && (
						<RichText
							className="cycle-blocks-profile__byline"
							multiline={ false }
							aria-label={ __( 'Post author byline text', 'cycle-blocks' ) }
							placeholder={ __( 'Write byline…', 'cycle-blocks' ) }
							value={ byline }
							onChange={ ( value ) =>
								setAttributes( { byline: value } )
							}
						/>
					) }

					{ authorDetails?.name && (
						<p className="cycle-blocks-profile__name">
							{ authorDetails?.name || __( 'Post Author', 'cycle-blocks' ) }
						</p>
					) }
					{ authorDetails?.description && showBio && (
						<p className="cycle-blocks-profile__bio">
							{ authorDetails?.description }
						</p>
					) }
				</div>
				{ showRecentPosts && latestPosts && (
					<div className="cycle-blocks-profile__recent-post">
						{ ( ! RichText.isEmpty( authorTitle ) || isSelected ) && (
							<RichText
								className="cycle-blocks-profile__author_title"
								tagName={ "h4" }
								multiline={ false }
								aria-label={ __( 'Author title', 'cycle-blocks' ) }
								placeholder={ __( 'Write title…', 'cycle-blocks' ) }
								value={ authorTitle }
								onChange={ ( value ) =>
									setAttributes( { authorTitle: value } )
								}
							/>
						) }
							<ul className={ post_class }>
								{ latestPosts.map( ( post, i ) => {
									const titleTrimmed = invoke( post, [
										'title',
										'rendered',
										'trim',
									] );

									const {
										featuredImageInfo: {
											url: imageSourceUrl,
											alt: featuredImageAlt,
										} = {},
									} = post;

									const renderFeaturedImage =
										displayFeaturedImage && ( imageSourceUrl || mediaSizes );

									const image_size = postLayout === 'grid' ? 'medium' : 'thumbnail';

									const featuredImage = renderFeaturedImage && (
										<img
											src={
												imageSourceUrl ? imageSourceUrl
												: mediaSizes[ image_size ] ? mediaSizes[ image_size ].source_url
												: mediaSizes[ 'full' ].source_url ? mediaSizes[ 'full' ].source_url
												: ''
											}
											alt={ featuredImageAlt }
											className={ `attachment-${ image_size } wp-post-image` }
										/>
									);

									return (
										<li key={ i }>
											{ displayFeaturedImage && featuredImage && (
												<div className="cycle-blocks-profile_recent-post__featured-image">
													<a href={ post.link } rel="noreferrer noopener">{ featuredImage }</a>
												</div>
											) }
											<div className="cycle-blocks-profile_recent-post__content">
												{ post.date_gmt && (
													<time
														dateTime={ format( 'c', post.date_gmt ) }
														className="cycle-blocks-profile__post-date"
													>
														{ dateI18n( dateFormat, post.date_gmt ) }
													</time>
												) }
												<a href={ post.link } rel="noreferrer noopener">
													{ titleTrimmed ? (
														<RawHTML>{ titleTrimmed }</RawHTML>
													) : (
														__( '(no title)', 'cycle-blocks' )
													) }
												</a>
											</div>
										</li>
									);
							} ) }
						</ul>
					</div>
				) }
			</div>
		</>
	);
}
