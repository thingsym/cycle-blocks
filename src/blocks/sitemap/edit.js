'use strict';

/**
 * WordPress dependencies
 */
import {
	Disabled,
	PanelBody,
	ToggleControl,
} from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
import ServerSideRender from '@wordpress/server-side-render';

export default function sitemapEdit( { attributes, setAttributes } ) {
	const {
		displayPage,
		displayPost,
		displayCategory,
		displayAuthor,
	} = attributes;

	const inspectorControls = (
		<InspectorControls>
			<PanelBody title={ __( 'Sitemap settings', 'cycle-blocks' ) } >
				<ToggleControl
					label={ __( 'Display Page', 'cycle-blocks' ) }
					checked={ displayPage }
					onChange={ () =>
						setAttributes( {
							displayPage: !displayPage,
						} )
					}
				/>
				<ToggleControl
					label={ __( 'Display Post', 'cycle-blocks' ) }
					checked={ displayPost }
					onChange={ () =>
						setAttributes( {
							displayPost: !displayPost,
						} )
					}
				/>
				<ToggleControl
					label={ __( 'Display Category', 'cycle-blocks' ) }
					checked={ displayCategory }
					onChange={ () =>
						setAttributes( {
							displayCategory: !displayCategory,
						} )
					}
				/>
				<ToggleControl
					label={ __( 'Display Author', 'cycle-blocks' ) }
					checked={ displayAuthor }
					onChange={ () =>
						setAttributes( {
							displayAuthor: !displayAuthor,
						} )
					}
				/>
			</PanelBody>
		</InspectorControls>
	);

	return (
		<>
			{ inspectorControls }
			<div { ...useBlockProps() }>
				<Disabled>
					<ServerSideRender
						block="cycle-blocks/sitemap"
						attributes={ attributes }
					/>
				</Disabled>
			</div>
		</>
	);
}
