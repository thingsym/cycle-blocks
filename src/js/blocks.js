'use strict';

/**
 * WordPress dependencies
 */
import {
	registerBlockType,
	unstable__bootstrapServerSideBlockDefinitions,
} from '@wordpress/blocks';
import { __, _x } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import * as cycle_blocks_page_list from '../blocks/page-list/index.js';
import * as cycle_blocks_sitemap from '../blocks/sitemap/index.js';
import * as cycle_blocks_profile from '../blocks/profile/index.js';
import * as cycle_blocks_fontawesome_icons from '../blocks/fontawesome-icons/index.js';

/**
 * Function to register an individual block.
 *
 * @param {Object} block The block to be registered.
 *
 */
const registerBlock = ( block ) => {
	if ( ! block ) {
		return;
	}

	let { metadata, settings, name } = block;

	if ( metadata ) {
		// for ServerSide Blocks
		unstable__bootstrapServerSideBlockDefinitions({ [name]: metadata });
	}

	[ metadata, settings ] = applyTextdomainMetadata( metadata, settings );

	registerBlockType( { name, ...metadata }, settings );
};

const applyTextdomainMetadata = ( metadata, settings ) => {
	if ( metadata ) {
		if ( !! metadata.title ) {
			metadata.title = _x( metadata.title, 'block title', 'cycle-blocks' );
			settings.title = metadata.title;
		}
		if ( !! metadata.description ) {
			metadata.description = _x( metadata.description, 'block description', 'cycle-blocks' );
			settings.description = metadata.description;
		}
		if ( !! metadata.keywords ) {
			metadata.keywords = __( metadata.keywords, 'cycle-blocks' );
			settings.keywords = metadata.keywords;
		}
	}

	return [ metadata, settings ];
}

[
	// Common blocks are grouped at the top to prioritize their display
	// in various contexts â€” like the inserter and auto-complete components.
	cycle_blocks_page_list,
	cycle_blocks_sitemap,
	cycle_blocks_profile,

	cycle_blocks_fontawesome_icons,

].forEach( registerBlock );
