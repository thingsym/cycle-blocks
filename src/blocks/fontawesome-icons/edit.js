'use strict';

/**
 * External dependencies
 */
import classnames from 'classnames';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';
import {
	PanelBody,
	BaseControl,
	ToggleControl,
} from '@wordpress/components';
import {
	InspectorControls,
	FontSizePicker,
	PanelColorSettings,
	ContrastChecker,
	BlockControls,
	AlignmentToolbar,
	useBlockProps,
} from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import IconSelectControl from '../../helper/icon-select-control/index.js';
import { IconSettings } from './icon-settings.js';

export default function fontawesomeIconsEdit( { attributes, setAttributes } ) {
	const {
		iconClassName,
		iconSize,
		enableInlineBlock,
		iconColor,
		backgroundColor,
		textAlign,
	} = attributes;

	const innerClassName = classnames( `${ iconClassName }` );

	const iconStyles = {
		fontSize: iconSize ? iconSize : null,
		color: iconColor || null,
	};

	const onChangeIconSize = ( value ) =>
		setAttributes( {
			iconSize: value,
		}
	);

	const onChangeBackgroundColor = ( value ) =>
		setAttributes( {
			backgroundColor: value,
		}
	);

	const onChangeIconColor = ( value ) =>
		setAttributes( {
			iconColor: value,
		}
	);

	const blockProps = useBlockProps( {
		className: classnames( {
			[ `has-text-align-${ textAlign }` ]: textAlign,
			[ `has-inline-icon` ]: enableInlineBlock,
		} ),
		style: {
			backgroundColor: backgroundColor || null,
			padding: enableInlineBlock && iconSize ? iconSize : null,
		}
	} );

	return (
		<div { ...blockProps }>
			<InspectorControls>
				<PanelBody
					title={ __(
						'Icon settings',
						'cycle-blocks'
					)}
				>

					<ToggleControl
						label={ __( 'Enable inline block', 'editor-bridge' ) }
						value={ enableInlineBlock }
						checked={ enableInlineBlock }
						onChange={ () => {
							setAttributes( {
								enableInlineBlock: ! enableInlineBlock
							} );
						} }
					/>

					<BaseControl
						label={ __( 'Icon Size', 'cycle-blocks' ) }
					>
						<FontSizePicker
							value={ iconSize }
							onChange={ onChangeIconSize }
						/>
					</BaseControl>
					<IconSelectControl
						label={ __( 'Icons', 'cycle-blocks' ) }
						valueType="class"
						value={ iconClassName }
						options={ IconSettings }
						onChange={ ( value ) => {
							setAttributes( {
								iconClassName: value,
							} );
						} }
					/>
				</PanelBody>
				<PanelColorSettings
					title={ __( 'Color Settings', 'cycle-blocks' ) }
					initialOpen={ false }
					colorSettings={ [
						{
							value: iconColor,
							onChange: onChangeIconColor,
							label: __( 'Icon Color', 'cycle-blocks' ),
						},
						{
							value: backgroundColor,
							onChange: onChangeBackgroundColor,
							label: __(
								'Background Color',
								'cycle-blocks'
							),
						},
					] }
				>
					<ContrastChecker
						backgroundColor={ backgroundColor }
						iconColor={ iconColor }
					/>
				</PanelColorSettings>
			</InspectorControls>

			<BlockControls>
				<AlignmentToolbar
					value={ textAlign }
					onChange={ ( nextAlign ) => {
						setAttributes( { textAlign: nextAlign } );
					} }
				/>
			</BlockControls>

			<i
				className={ innerClassName }
				style={ iconStyles }
			/>
		</div>
	);

}
