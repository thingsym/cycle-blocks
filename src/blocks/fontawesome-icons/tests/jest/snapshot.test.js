/**
* External dependencies
*/
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

/**
 * Internal dependencies
 */
import { default as fontawesomeIconsEdit } from '../../edit.js';

describe( 'fontawesomeIconsEdit snapshot', () => {
	it( 'fontawesomeIconsEdit snapshot', () => {
		const elem = shallowToJson( shallow( <fontawesomeIconsEdit /> ) );
		expect( elem ).toMatchSnapshot();
	} );
} );
