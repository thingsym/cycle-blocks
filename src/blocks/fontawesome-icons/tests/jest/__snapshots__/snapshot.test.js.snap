// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontawesomeIconsEdit snapshot fontawesomeIconsEdit snapshot 1`] = `<fontawesomeIconsEdit />`;
